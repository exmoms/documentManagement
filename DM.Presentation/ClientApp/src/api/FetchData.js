/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
