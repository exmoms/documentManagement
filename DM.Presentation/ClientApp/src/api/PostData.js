/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
