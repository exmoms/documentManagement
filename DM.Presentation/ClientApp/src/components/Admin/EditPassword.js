import React, { useState } from "react";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import InputLabel from "@material-ui/core/InputLabel";
// import { postData } from "../../api/PostData";
import { postDataToAPI } from "../../api/PostData";
import { useTranslation } from "react-i18next";
import clsx from "clsx";
import InputAdornment from "@material-ui/core/InputAdornment";
import IconButton from "@material-ui/core/IconButton";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import FormControl from "@material-ui/core/FormControl";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { Notification } from "../Admin/Notifications";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
}));

export default function EditPassword() {
  const { t } = useTranslation();
  const classes = useStyles();

  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [values, setValues] = useState({
    password: "",
    showPassword: false,
  });

  const regester_file = {
      NewPassword: values.password,
      CurrentPassword: password,
      ConfirmPassword: confirmPassword,
  };

  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);
  const [open, setOpen] = React.useState(false);

  const onchange = (event) => {
    if (event.target.name === "password") {
      setPassword(event.target.value);
    } else if (event.target.name === "confirmPassword") {
      setConfirmPassword(event.target.value);
    }
  };

  const handelSubmit = async (event) => {
    event.preventDefault();
    if (password.length < 8 || password.length > 12) {
      seterror(true);
      seterrorMessage([t("password_charachters_no")]);
      setOpen(true);
    } else if (values.password.length < 8 || confirmPassword.length > 12) {
      seterror(true);
      seterrorMessage([t("password_charachters_no")]);
      setOpen(true);
    } else if (values.password !== confirmPassword) {
      seterror(true);
      seterrorMessage([t("password_mismatch")]);
      setOpen(true);
    } else {
      let response = await postDataToAPI(
        "/api/user/EditPassword",
        regester_file
      );
      if (response.ok) {
        setOpen(true);
      } else {
        let json = await response.json();
        seterror(true);
        seterrorMessage(json.error);
        setOpen(true);
      }
    }
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickShowPassword = () => {
    setValues({ ...values, showPassword: !values.showPassword });
  };

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <form className={classes.form} onSubmit={handelSubmit} id="submitForm">
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="password"
                label={t("password")}
                type="password"
                id="password"
                autoComplete="current-password"
              />
            </Grid>

            <FormControl
              className={clsx(classes.margin, classes.textField)}
              variant="outlined"
            >
              <InputLabel htmlFor="outlined-adornment-password">
                {t("new_password")}
              </InputLabel>
              <OutlinedInput
                id="outlined-adornment-password"
                type={values.showPassword ? "text" : "password"}
                value={values.password}
                onChange={handleChange("password")}
                endAdornment={
                  <InputAdornment position="end">
                    <IconButton
                    id="iconButton1"
                      aria-label="toggle password visibility"
                      onClick={handleClickShowPassword}
                      onMouseDown={handleMouseDownPassword}
                      edge="end"
                    >
                      {values.showPassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                }
                labelWidth={70}
              />
            </FormControl>

            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="confirmPassword"
                label={t("confirmNewPassword_label")}
                type="password"
                id="confirmPassword"
                autoComplete="current-password"
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            {t("buttton_save")}
          </Button>
        </form>
      </div>
      <Notification
      id="notification"
        open={open} // must be a "state" to show notification message [true or false]
        setOpen={setOpen} // this function changes the state "open" value.
        error={error} // if error message put [true] else for success put [false]
        errorMessage={errorMessage} // the recived error message (array of string) from server.
      />
    </Container>
  );
}
