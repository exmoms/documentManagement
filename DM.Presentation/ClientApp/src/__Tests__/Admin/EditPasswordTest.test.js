
import React from "react";
import ReactDom from "react-dom";
import EditPassword from "../../components/Admin/EditPassword";
import { cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow , mount} from "enzyme";
import toJSON from "enzyme-to-json";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Link } from "react-router-dom";
import { Suspense } from "react";
import * as FetchMock from "../../api/FetchData";
import * as PostDataMock from "../../api/PostData";
import * as ContextAuth from "../../context/auth";
import MenuItem from "@material-ui/core/MenuItem";
import { act } from "react-dom/test-utils";
import List from "@material-ui/core/List";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
afterEach(cleanup);
describe("EditPassword", () => {
   afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(
        <EditPassword/>,
      div
    );
  });

  test("matches snapshot", () => {
    const wrapper = shallow(
         <EditPassword />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("intialization AppSideBar when role ia admin", () => {
    let  wrapper = mount(<EditPassword />);
    expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("handleClickShowPassword", () => {
    let  wrapper = mount(<EditPassword />);
    act(() => {
        wrapper
          .find("#iconButton1")
          .at(0)
          .props()
          .onClick();
      });
      wrapper.render();
        wrapper.update();
    expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("text");
    expect(wrapper.find(Visibility).length).toBe(1);
    expect(wrapper.find(VisibilityOff).length).toBe(0);
    expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("handleMouseDownPassword", () => {
    let  wrapper = mount(<EditPassword />);
    const preventDefaultMock = jest.fn();
    act(() => {
        wrapper
          .find("#iconButton1")
          .at(0)
          .props()
          .onMouseDown({ preventDefault:preventDefaultMock});
      });
      wrapper.render();
        wrapper.update();
        expect(preventDefaultMock).toHaveBeenCalledTimes(1);
        expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
        expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
        expect(wrapper.find(Visibility).length).toBe(0);
        expect(wrapper.find(VisibilityOff).length).toBe(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("setOpen", () => {
    let  wrapper = mount(<EditPassword />);
    expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
        expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
        expect(wrapper.find(Visibility).length).toBe(0);
        expect(wrapper.find(VisibilityOff).length).toBe(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
    act(() => {
        wrapper
          .find("#notification")
          .at(0)
          .props()
          .setOpen(true);
      });
      wrapper.render();
        wrapper.update();
        expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
        expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
        expect(wrapper.find(Visibility).length).toBe(0);
        expect(wrapper.find(VisibilityOff).length).toBe(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("handleChange", () => {
    let  wrapper = mount(<EditPassword />);
    expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("");
        expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
        expect(wrapper.find(Visibility).length).toBe(0);
        expect(wrapper.find(VisibilityOff).length).toBe(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
    act(() => {
        wrapper
          .find("#outlined-adornment-password")
          .at(0)
          .props()
          .onChange({ target: { value:  "password" } });
      });
      wrapper.render();
        wrapper.update();
        expect(wrapper.find("#outlined-adornment-password").at(0).props().value).toBe("password");
        expect(wrapper.find("#outlined-adornment-password").at(0).props().type).toBe("password");
        expect(wrapper.find(Visibility).length).toBe(0);
        expect(wrapper.find(VisibilityOff).length).toBe(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("onchange and handleSubmit when no error", () => {
    let  wrapper = mount(<EditPassword />);
    const preventDefaultMock = jest.fn();
    const logMock=  jest.spyOn(console, "log")
    //first branch when name:"password"
    act(() => {
        wrapper
          .find("#password")
          .at(0)
          .props()
          .onChange({ target: { value: "passwordA" ,name:"password"} });
      });
        wrapper.update();
         //second branch when name:"confirmPassword"
        act(() => {
            wrapper
              .find("#password")
              .at(0)
              .props()
              .onChange({ target: { value: "passwordv" ,name:"confirmPassword"} });
          });
          //setting Values.password
          act(() => {
            wrapper
              .find("#outlined-adornment-password")
              .at(0)
              .props()
              .onChange({ target: { value:  "passwordv" } });
          });
            wrapper.update();
            //calling submit help expect the setted values.
            act(() => {
                wrapper
                  .find("#submitForm")
                  .at(0)
                  .props()
                  .onSubmit({ preventDefault:preventDefaultMock});
              });
                wrapper.update();
                expect(preventDefaultMock).toHaveBeenCalledTimes(1);
                expect(wrapper.find(Visibility).length).toBe(0);
                expect(wrapper.find(VisibilityOff).length).toBe(1);
                expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
              expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
              expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
  });

  test("handleSubmit when (password.length < 8 || password.length > 12)", () => {
    let  wrapper = mount(<EditPassword />);
    const mockSuccessResponse ={ok :false};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const PostMock = jest.spyOn(PostDataMock, "postDataToAPI").mockReturnValue(mockJsonPromise);
    const preventDefaultMock = jest.fn();
    const logMock=  jest.spyOn(console, "log")
    expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
    expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
   expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
    act(() => {
        wrapper
          .find("#password")
          .at(0)
          .props()
          .onChange({ target: { value: "unValidPassword" ,name:"password"} });
      });
        wrapper.update();
            //calling submit help expect the setted values.
            return mockJsonPromise.then(() => {
            act(() => {
                wrapper
                  .find("#submitForm")
                  .at(0)
                  .props()
                  .onSubmit({ preventDefault:preventDefaultMock});
              });
            }).then(() => {
              wrapper.update();
              wrapper.render();
                expect(preventDefaultMock).toHaveBeenCalledTimes(1);
                expect(logMock).toHaveBeenCalledTimes(0);
                expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
                expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
                expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["password_charachters_no"]);
  });
});

  // test("onchange and handleSubmit when (values.password.length < 8 || confirmPassword.length > 12)", () => {
  //   const mockSuccessResponse ={ok :true};
  //   const mockJsonPromise = Promise.resolve(mockSuccessResponse);
  //   global.fetch = require('jest-fetch-mock');
  //   fetch.mockResponse(mockJsonPromise);
  //   let  wrapper = mount(<EditPassword />);
  //   const preventDefaultMock = jest.fn();
  //   const logMock=  jest.spyOn(console, "log")
  //   //first branch when name:"password"
  //   act(() => {
  //       wrapper
  //         .find("#password")
  //         .at(0)
  //         .props()
  //         .onChange({ target: { value: "passwordA" ,name:"password"} });
  //     });
  //       wrapper.update();
  //        //second branch when name:"confirmPassword"
  //       act(() => {
  //           wrapper
  //             .find("#password")
  //             .at(0)
  //             .props()
  //             .onChange({ target: { value: "unValidConfirm" ,name:"confirmPassword"} });
  //         });
  //         //setting Values.password
  //         act(() => {
  //           wrapper
  //             .find("#outlined-adornment-password")
  //             .at(0)
  //             .props()
  //             .onChange({ target: { value:  "passwordv" } });
  //         });
  //           wrapper.update();
  //           //calling submit help expect the setted values.
  //           act(() => {
  //               wrapper
  //                 .find("#submitForm")
  //                 .at(0)
  //                 .props()
  //                 .onSubmit({ preventDefault:preventDefaultMock});
  //             });
  //               wrapper.update();
  //               expect(preventDefaultMock).toHaveBeenCalledTimes(1);
  //               expect(logMock).toHaveBeenCalledTimes(0);
  //               expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
  //               expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
  //               expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["password_charachters_no"]);
  // });

  // test("onchange and handleSubmit when  (values.password !== confirmPassword)", () => {
  //   const mockSuccessResponse ={ok :true};
  //   const mockJsonPromise = Promise.resolve(mockSuccessResponse);
  //   global.fetch = require('jest-fetch-mock');
  //   fetch.mockResponse(mockJsonPromise);
  //   let  wrapper = mount(<EditPassword />);
  //   const preventDefaultMock = jest.fn();
  //   const logMock=  jest.spyOn(console, "log")
  //   //first branch when name:"password"
  //   act(() => {
  //       wrapper
  //         .find("#password")
  //         .at(0)
  //         .props()
  //         .onChange({ target: { value: "passwordA" ,name:"password"} });
  //     });
  //       wrapper.update();
  //        //second branch when name:"confirmPassword"
  //       act(() => {
  //           wrapper
  //             .find("#password")
  //             .at(0)
  //             .props()
  //             .onChange({ target: { value: "passwordW" ,name:"confirmPassword"} });
  //         });
  //         //setting Values.password
  //         act(() => {
  //           wrapper
  //             .find("#outlined-adornment-password")
  //             .at(0)
  //             .props()
  //             .onChange({ target: { value:  "passwordv" } });
  //         });
  //           wrapper.update();
  //           //calling submit help expect the setted values.
  //           act(() => {
  //               wrapper
  //                 .find("#submitForm")
  //                 .at(0)
  //                 .props()
  //                 .onSubmit({ preventDefault:preventDefaultMock});
  //             });
  //               wrapper.update();
  //               expect(logMock).toHaveBeenCalledTimes(0);
  //               expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
  //               expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
  //               expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["password_mismatch"]);
  // });
});