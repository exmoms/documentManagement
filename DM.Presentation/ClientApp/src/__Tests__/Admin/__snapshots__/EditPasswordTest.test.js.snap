// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPassword matches snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <WithStyles(CssBaseline) />
  <div
    className="makeStyles-paper-1"
  >
    <form
      className="makeStyles-form-2"
      id="submitForm"
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="current-password"
            fullWidth={true}
            id="password"
            label="password"
            name="password"
            onChange={[Function]}
            type="password"
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(FormControl))
          className="makeStyles-margin-4 makeStyles-textField-5"
          variant="outlined"
        >
          <WithStyles(ForwardRef(InputLabel))
            htmlFor="outlined-adornment-password"
          >
            new_password
          </WithStyles(ForwardRef(InputLabel))>
          <WithStyles(ForwardRef(OutlinedInput))
            endAdornment={
              <ForwardRef(WithStyles)
                position="end"
              >
                <ForwardRef(WithStyles)
                  aria-label="toggle password visibility"
                  edge="end"
                  id="iconButton1"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                >
                  <UNDEFINED />
                </ForwardRef(WithStyles)>
              </ForwardRef(WithStyles)>
            }
            id="outlined-adornment-password"
            labelWidth={70}
            onChange={[Function]}
            type="password"
            value=""
          />
        </WithStyles(ForwardRef(FormControl))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="current-password"
            fullWidth={true}
            id="confirmPassword"
            label="confirmNewPassword_label"
            name="confirmPassword"
            onChange={[Function]}
            type="password"
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-3"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        buttton_save
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
  <Notification
    error={false}
    errorMessage={Array []}
    id="notification"
    open={false}
    setOpen={[Function]}
  />
</WithStyles(ForwardRef(Container))>
`;
